src/impy/models/pythia8.py

```diff
@@ -64,11 +61,12 @@ def __init__(self, evt_kin, *, seed=None):
             del environ["PYTHIA8DATA"]
         self._lib.pythia = self._lib.Pythia(datdir, True)
 
-        # must come last
-        self.kinematics = evt_kin
-        self._set_final_state_particles()
-
-    def _cross_section(self, kin=None):
+    def _cross_section(self, kin):
+        if (kin.p2.A or 1) > 1:
```

Comment:
It's really that bad at nuclei :)

Reply:
I don't know, perhaps you have to query the cross-section from another variable. At some point I dereference a nullptr in the calls when nuclei are used. I  didn't check what exactly happens.